src/sql/workbench/contrib/query/browser/keyboardQueryActions.ts

```diff
@@ -131,6 +133,76 @@ export class RunCurrentQueryKeyboardAction extends Action {
 	}
 }
 
+export class CopyQueryWithResultsKeyboardAction extends Action {
+	public static ID = 'copyQueryWithResultsKeyboardAction';
+	public static LABEL = nls.localize('copyQueryWithResultsKeyboardAction', "Copy Query With Results");
+
+	constructor(
+		id: string,
+		label: string,
+		@IEditorService private _editorService: IEditorService,
+		@IClipboardService private _clipboardService: IClipboardService,
+		@IQueryModelService protected readonly queryModelService: IQueryModelService,
+	) {
+		super(id, label);
+		this.enabled = true;
+	}
+
+	public async getFormattedResults(editor): Promise<any> {
+		let queryRunner = this.queryModelService.getQueryRunner(editor.input.uri);
+		let allResults = '';
+		let allHtmlResults = '';
+
+		if (queryRunner && queryRunner.batchSets.length > 0) {
+			for (let i = 0; i < queryRunner.batchSets[0].resultSetSummaries.length; i++) {
+				let resultSummary = queryRunner.batchSets[0].resultSetSummaries[i];
+				let result = await queryRunner.getQueryRows(0, resultSummary.rowCount, resultSummary.batchId, resultSummary.id);
+				let tableHeaders = resultSummary.columnInfo.map((col, i) => (col.columnName));
+				let htmlTableHeaders = resultSummary.columnInfo.map((col, i) => (`<th style="border:solid black 1.0pt; whiteSpace:nowrap">${escape(col.columnName)}</th>`));
+				let copyString = '\n';
+				let htmlCopyString = '<tr>';
+
+				for (let rowEntry of result.rows) {
+					for (let colIdx = 0; colIdx < rowEntry.length; colIdx++) {
+						let value = rowEntry[colIdx].displayValue;
+						if (value) {
+							copyString = `${copyString}${value}\t`;
+							htmlCopyString = `${htmlCopyString}<td style="border:solid black 1.0pt;white-space:nowrap">${escape(value)}</td>`;
+						}
+					}
+					// Removes the tab seperator from the end of a row
+					copyString = copyString.slice(0, -1 * '\t'.length) + '\n';
+					htmlCopyString = htmlCopyString + '</tr>';
+				}
+
+				allResults = `${allResults}${tableHeaders.join('\t')}${copyString}\n`;
+				allHtmlResults = `${allHtmlResults}<div><br/><br/>
+				<table cellPadding="5" cellSpacing="1" style="border:1;border-color:Black;font-family:Segoe UI;font-size:12px;border-collapse:collapse">
+				<tr style="background-color:DarkGray">${htmlTableHeaders.join('')}</tr>${htmlCopyString}
+				</table></div>`;
+			}
+		}
+
+		return { result: allResults, html: allHtmlResults };
+	}
+
+	public async run(): Promise<void> {
+		const editor = this._editorService.activeEditorPane;
+		if (editor instanceof QueryEditor) {
+			let allResults = await this.getFormattedResults(editor);
+			let queryText = editor.getAllText();
+
+			let data = {
+				text: `${queryText}\n\n${allResults.result}`,
+				html: `${escape(queryText).replace(/\r\n|\n|\r/gm, '<br />')}${allResults.html}`
+			};
+
+			this._clipboardService.write(data);
+		}
+		return Promise.resolve(null);
```

Comment:
1. You shouldn't be returning null - we use undefined in nearly all cases
1. The return type is void so you shouldn't be returning anything at all
1. Since this is an async function you can just remove this line and it'll work as expected

Reply:
For the above comment related to Outlook, notepad behavior: 
Yes correct. Excel, Outlook supports HTML text, it shows up with all the formatting whereas Notepad doesnt, so its just plain text.